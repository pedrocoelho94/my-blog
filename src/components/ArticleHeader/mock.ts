import { data } from 'api/mockPosts.json'

const { id, title, excerpt, cover, author, categories, createdAt } =
  data.posts[0]

export default {
  id,
  title,
  excerpt,
  cover,
  author,
  categories,
  createdAt
}
