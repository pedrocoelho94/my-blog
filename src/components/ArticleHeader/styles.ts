import styled, { css } from 'styled-components'
import * as HeadingStyles from 'components/Heading/styles'

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles.Title} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
      font-size: ${theme.font.sizes.large};
    }
  `}
`

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: calc(${theme.font.sizes.medium} * 0.9);
  `}
`

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`
