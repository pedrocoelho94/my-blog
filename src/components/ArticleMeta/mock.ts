export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  author: {
    id: '123456789',
    slug: 'pedro-coelho',
    displayName: 'Pedro Coelho'
  },
  categories: [
    {
      id: '78724ec23f',
      displayName: 'Tech',
      slug: 'tech'
    },
    {
      id: '762vg214g34',
      displayName: 'JS',
      slug: 'javascript'
    }
  ]
}
