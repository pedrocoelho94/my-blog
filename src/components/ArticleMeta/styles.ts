import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span:not(:last-child):after {
      content: ', ';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 0.2s ease-in-out;

      &:hover {
        filter: brightness(0.5);
      }
    }
  `}
`
