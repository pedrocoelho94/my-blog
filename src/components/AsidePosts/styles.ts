import styled, { css } from 'styled-components'

export const Wrapper = styled.div``

export const Content = styled.a`
  ${({ theme }) => css`
    display: flex;
    gap: 1rem;
    height: 100%;
    cursor: pointer;
    text-decoration: none;
    margin: ${theme.spacings.medium} 0;
  `}
`

export const Cover = styled.img`
  border-radius: 0.8rem;
  width: 10rem;
  height: 10rem;
  object-fit: cover;
`

export const Title = styled.h4`
  ${({ theme }) => css`
    font-size: 1.5rem;
    height: 100%;
    margin: 0;
    color: ${theme.colors.darker};

    &:hover {
      text-decoration: underline;
    }
  `}
`
