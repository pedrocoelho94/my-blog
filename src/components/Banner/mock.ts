import { data } from 'api/mockPosts.json'

export default {
  settings: data.setting,
  post: data.posts[0]
}
