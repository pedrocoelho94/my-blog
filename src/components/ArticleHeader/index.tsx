import ArticleMeta, { ArticleMetaProps } from 'components/ArticleMeta'
import Heading from 'components/Heading'
import { StrapiImage } from 'shared-typed/cover'
import * as S from './styles'

export type ArticleHeaderProps = {
  id: string
  title: string
  excerpt: string
  cover: StrapiImage
} & ArticleMetaProps

const ArticleHeader = ({
  title,
  excerpt,
  cover,
  author,
  categories,
  createdAt
}: ArticleHeaderProps) => (
  <S.Wrapper>
    <Heading size="big">{title}</Heading>
    <S.Excerpt>{excerpt}</S.Excerpt>
    <S.Cover src={cover.url} alt={cover.alternativeText} />
    <ArticleMeta
      author={author}
      categories={categories}
      createdAt={createdAt}
    />
  </S.Wrapper>
)

export default ArticleHeader
