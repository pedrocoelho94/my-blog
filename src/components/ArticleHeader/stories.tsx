import { Story, Meta } from '@storybook/react'
import ArticleHeader, { ArticleHeaderProps } from '.'

import mock from './mock'

export default {
  title: 'Posts/ArticleHeader',
  component: ArticleHeader,
  args: mock
} as Meta

export const Default: Story<ArticleHeaderProps> = (args) => (
  <div>
    <ArticleHeader {...args} />

    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Deserunt, ea
      tempore doloribus laboriosam tenetur aperiam dolorum rem repudiandae
      adearum. Praesentium quae totam nobis aliquid cum modi eos est doloremque.
    </p>

    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Repudiandae
      excepturi modi sint odit laudantium illum inventore aliquam consequuntur
      illo dolor, debitis provident adipisci, culpa vero nisi saepe tempora
      veritatis? Mollitia. Lorem ipsum, dolor sit amet consectetur adipisicing
      elit. Dolorem, dignissimos. Dicta ea cumque sunt, animi deleniti
      blanditiis numquam consequatur dolores saepe illum neque ipsam explicabo
      maxime, labore at provident officiis.
    </p>
  </div>
)
