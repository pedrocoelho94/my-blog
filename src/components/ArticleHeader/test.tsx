import { screen } from '@testing-library/react'
import { renderWithTheme } from 'styles/renderWithTheme'
import { formatDate } from 'utils/format-date'

import ArticleHeader, { ArticleHeaderProps } from '.'
import mock from './mock'

const props: ArticleHeaderProps = mock

describe('<ArticleHeader />', () => {
  it('should render heading, excerpt, cover img and meta', () => {
    const { container } = renderWithTheme(<ArticleHeader {...props} />)

    expect(
      screen.getByRole('heading', { name: props.title })
    ).toBeInTheDocument()

    expect(screen.getByText(props.excerpt)).toBeInTheDocument()

    expect(
      screen.getByRole('img', { name: props.cover.alternativeText })
    ).toBeInTheDocument()

    expect(screen.getByText(formatDate(props.createdAt))).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
