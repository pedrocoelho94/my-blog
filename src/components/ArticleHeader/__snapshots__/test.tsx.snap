// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<ArticleHeader /> should render heading, excerpt, cover img and meta 1`
] = `
.c5 {
  font-size: 1.6rem;
  color: #AAAAAA;
  font-style: italic;
}

.c5 .categories span:not(:last-child):after {
  content: ', ';
}

.c5 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c5 a:hover {
  -webkit-filter: brightness(0.5);
  filter: brightness(0.5);
}

.c2 {
  color: #000000;
  line-height: 1.2;
  font-size: 4.0rem;
  text-transform: none;
}

.c0 {
  padding-bottom: 4.0rem;
  margin-bottom: 4.0rem;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c0 .c1 {
  margin: 0;
  margin-bottom: 2.4rem;
  font-size: 3.2rem;
}

.c3 {
  margin: 2.4rem 0;
  font-size: calc(2.4rem * 0.9);
}

.c4 {
  max-width: 100%;
  display: block;
  margin-bottom: 2.4rem;
}

@media (max-width:768px) {
  .c2 {
    font-size: calc(1.6rem * 1.2);
  }
}

<div>
  <header
    class="c0"
  >
    <h1
      class="c1 c2"
    >
      Al Sadd anuncia acordo para liberar Xavi, mas Barcelona nega
    </h1>
    <p
      class="c3"
    >
      A novela continua: Barcelona vê em anúncio de acerto do Al Sadd como uma forma de pressão para o pagamento da multa.
    </p>
    <img
      alt=""
      class="c4"
      src="https://res.cloudinary.com/rabbitz/image/upload/v1636129235/Xavi_Hernandez_Barcelona_c60f5b3e16.jpg"
    />
    <div
      class="c5"
    >
      <p>
        <span>
          Por 
        </span>
        <a
          href="/author/pedro-coelho"
        >
          Pedro Coelho
        </a>
        <span
          class="separator"
        >
           em 
        </span>
        <time
          datetime="2020-04-20"
        >
          19 de abr. de 2020
        </time>
        <span
          class="separator"
        >
           - 
        </span>
        <span
          class="categories"
        >
          <span>
            <a
              href="/category/futebol"
            >
              Futebol
            </a>
          </span>
        </span>
      </p>
    </div>
  </header>
</div>
`
